### PR TITLE
serial/parallel nemsio.

### LINKED ISSUE #60: build nemsio w/ serial option if MPI is not found.
NEMSIO can be built serially, but is not.
Use cases are in programs utilizing codes such as `glbevents.f`

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
- allows the building of the nemsio library in a serial mode.

This is particularly useful when linking with codes that use only the serial nemsio capability e.g. `glbevents.f`.
This feature allows the developer to load modules under the compiler hierarchy, which implies the executable will be a serial executable.  Currently, this turns these execs into `mpirun -np 1 ./exec` which is less than desireable.

When MPI is found, it will default to its normal behaviour.

closes #60

### CODE REVIEW

**Comment by aerorahul** [comment]
CI is failing.  Need to fix the CI.

**Review by junwang-noaa — APPROVED** [review]

**Review by edwardhartnett — APPROVED** [review]

**Review by kgerheiser — APPROVED** [review]

File: cmake/PackageConfig.cmake.in [hunk 1]
```diff
@@ -15,7 +15,9 @@ include(CMakeFindDependencyMacro)
 
 find_dependency(bacio CONFIG)
 find_dependency(w3emc CONFIG)
-find_dependency(MPI COMPONENTS Fortran)
+if(@MPI_FOUND@)
```
Comment (kgerheiser) [inline cmake/PackageConfig.cmake.in]: Should this also be `MPI_Fortran_FOUND`?

Comment (aerorahul) [reply cmake/PackageConfig.cmake.in]: yes.

Comment (aerorahul) [reply cmake/PackageConfig.cmake.in]:
```suggestion
if(@MPI_Fortran_FOUND@)
```

**Pull request merged**